src/shogun/lib/parameter_observers/ObservedValue.h

```diff
@@ -69,10 +64,12 @@ namespace shogun
 		 * @param name param's name
 		 * @param value Any-wrapped value of the param
 		 */
-		ObservedValue(
-		    int64_t step, std::string& name, Any value, SG_OBS_VALUE_TYPE type)
-		    : m_step(step), m_name(name), m_value(value), m_type(type)
+		ObservedValue(int64_t step, std::string& name, Any value)
+		    : m_step(step), m_name(name), m_value(value)
 		{
+			SG_ADD(&m_step, "step", "Observation step", ParameterProperties::NONE);
+			SG_ADD(&m_name, "name", "Observed value name", ParameterProperties::NONE);
+			SG_ADD(&m_value, "value", "Observed value", ParameterProperties::NONE);
```

Comment:
This should enable `get` for the value observed. However, I fear that this is not the solution either since we would just be able to "get" again an `Any` value.